test/functional/atomic/test-atomic-forall-basic.cpp

```diff
@@ -0,0 +1,119 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Source file containing basic functional tests for atomic operations with forall.
+///
+
+#include "test-atomic-forall-basic.hpp"
+
+// top layer of function templates for tests
+template <typename ExecPolicy, typename AtomicPolicy>
+void testAtomicFunctionPol()
+{
+  testAtomicFunctionBasic<ExecPolicy, AtomicPolicy, int, 10000>();
```

Comment:
We might want to call the tests as TYPED_TESTS and pass in separate lists associated with AtomicFunctionBasic and AtomicFunctionLogical, this way we can get the types if a test fails in the future for any reason

Reply:
I sympathize with your desire to obtain the types upon failures, but it would be difficult to have a TYPED_TEST with a value parameter (10000, and 100000) in the template signature. The way to do what you're suggesting would be to have separate test cases for each value parameter (like AtomicFunctionBasic10000, etc.), but that is probably not tenable.

Reply:
I do see what you mean, I had just noticed that all `Func` tests are called with 10,000 and all `Logical` tests with 100,000 I imagined something more along the lines of :

`TYPED_TEST(AtomicFunc, OpenMP_auto_AtomicFuncFunctionalTest)`
`{`
`  testAtomicFunctionBasic<ExecPolicy, AtomicPolicy, TypeParam, 10000>();`
`}`
`TYPED_TEST(AtomicLogical, OpenMP_auto_AtomicLogicalFunctionalTest)`
`{`
`  testAtomicLogical<ExecPolicy, AtomicPolicy, TypeParam, 100000>();`
`}`